src/librustc_mir/check_const_err.rs

```diff
@@ -1,197 +0,0 @@
-use rustc::middle::const_val::{ConstVal, ConstEvalErr, ErrKind};
-use rustc::hir::def::Def;
-use rustc::traits;
-use interpret::eval_body_with_mir;
-use rustc::ty::{TyCtxt, ParamEnv};
-use rustc::ty::Instance;
-use rustc::ty::layout::LayoutOf;
-use rustc::hir::def_id::DefId;
-use rustc::ty::subst::Substs;
-
-fn is_const<'a, 'tcx: 'a>(tcx: TyCtxt<'a, 'tcx, 'tcx>, def_id: DefId) -> bool {
-    if let Some(node) = tcx.hir.get_if_local(def_id) {
-        match node {
-            Node::NodeItem(&hir::Item {
-                node: hir::ItemConst(..), ..
-            }) => true,
-            _ => false
-        }
-    } else {
-        match tcx.describe_def(def_id) {
-            Some(Def::Const(_)) => true,
-            _ => false
-        }
-    }
-}
-
-pub fn check<'a, 'tcx: 'a>(tcx: TyCtxt<'a, 'tcx, 'tcx>, def_id: DefId) {
-    let mir = &tcx.optimized_mir(def_id);
-    let substs = Substs::identity_for_item(tcx, def_id);
-    let instance = Instance::new(def_id, substs);
-    let param_env = tcx.param_env(def_id);
-
-    if is_const(tcx, def_id) {
-        let cid = GlobalId {
-            instance,
-            promoted: None,
-        };
-        eval_body_with_mir(tcx, cid, mir, param_env);
-    }
-
-    ConstErrVisitor {
-        tcx,
-        mir,
-    }.visit_mir(mir);
-    let outer_def_id = if tcx.is_closure(def_id) {
-        tcx.closure_base_def_id(def_id)
-    } else {
-        def_id
-    };
-    let generics = tcx.generics_of(outer_def_id);
-    // FIXME: miri should be able to eval stuff that doesn't need info
-    // from the generics
-    if generics.parent_types as usize + generics.types.len() > 0 {
-        return;
-    }
-    for i in 0.. mir.promoted.len() {
-        use rustc_data_structures::indexed_vec::Idx;
-        let cid = GlobalId {
-            instance,
-            promoted: Some(Promoted::new(i)),
-        };
-        eval_body_with_mir(tcx, cid, mir, param_env);
-    }
-}
-
-struct ConstErrVisitor<'a, 'tcx: 'a> {
-    tcx: TyCtxt<'a, 'tcx, 'tcx>,
-    mir: &'a Mir<'tcx>,
-}
-
-impl<'a, 'tcx> ConstErrVisitor<'a, 'tcx> {
-    fn eval_op(&self, op: &Operand<'tcx>) -> Option<u128> {
-        let op = match *op {
-            Operand::Constant(ref c) => c,
-            _ => return None,
-        };
-        match op.literal {
-            Literal::Value { value } => match value.val {
-                ConstVal::Value(Value::ByVal(PrimVal::Bytes(b))) => Some(b),
-                _ => return None,
-            },
-            _ => None,
-        }
-    }
-}
-
-impl<'a, 'tcx> Visitor<'tcx> for ConstErrVisitor<'a, 'tcx> {
-    fn visit_terminator(&mut self,
-                        block: BasicBlock,
-                        terminator: &Terminator<'tcx>,
-                        location: Location) {
-        self.super_terminator(block, terminator, location);
-        match terminator.kind {
-            TerminatorKind::Assert { ref cond, expected, ref msg, .. } => {
```

Comment:
Do you handle this in const prop now? I didn't realize what this file was, makes more sense to put it in const prop.

Reply:
yea I was doing that the whole time. Totally forgot about the const checker and trans version of that.